pkgs/development/python-modules/panda3d/binary-hashes.nix

```diff
@@ -0,0 +1,19 @@
+version : builtins.getAttr version {
```

Comment:
Can we build from source? Why are we not using fetchPypi? This should get an update script.

Reply:
Let me try to build from source. Thanks!

Reply:
I am not very familiar with how `fetchPypi` works though:

1. In this case, Pypi only provides `whl`, I think `fetchPypi` handle downloading the corresponding`wheel` with specific system?
2. Since Pypi only provides `whl` (I am not sure why), for building from source can I use `fetchFromGitHub`?

Reply:
yeah that's fine

Reply:
Thanks for the confirmation! Another question I need to consult:

This project uses a customized script to build: https://github.com/panda3d/panda3d/blob/master/makepanda/makepanda.py

And it will create a "wheel" file as the result. What I can do is

1. Set `format = "other"` in `buildPythonPackage`.
2. Invoke the custom build script in `buildPhase` to generate the "wheel" file.

I think there needs to be a 3rd step to unpack the wheel and create the usual `site-packages` directory in `$out`, but I am not sure what's the best approach to achieve that. Can you help advise? Thanks!

Reply:
that makefile looks pretty hairy
```
elif target == 'linux' and (os.path.isfile("/lib/libc-2.17.so") or os.path.isfile("/lib64/libc-2.17.so")) and os.path.isdir("/opt/python"):
```
If i had seen that, and if fetching the wheel like you do currently works, I don't think I personally would put energy into building it from source.

@SuperSandro2000 What do you think

Reply:
Generally though you can find information on what buildPythonPackage does in the nixpkgs manual: https://nixos.org/manual/nixpkgs/stable/#python

Reply:
Definitely. Thank you @dali99 !

Reply:
I would simply patch that check and replace it with stdenv.isLinux.